core/blockchain.go

```diff
@@ -838,6 +838,9 @@ func (bc *BlockChain) Stop() {
 		for _, offset := range []uint64{0, 1, TriesInMemory - 1} {
 			if number := bc.CurrentBlock().NumberU64(); number > offset {
 				recent := bc.GetBlockByNumber(number - offset)
+				if recent.Root() == (common.Hash{}) {
```

Comment:
AFAIK, block root can never be `common.Hash{}`, `56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421` is the empty root.